### PR TITLE
public/user listing studies

### PR DESCRIPTION
**Pull request by antgonza**
This solves an issue introduced in the initial listing PR, adds 2 lists with only one control set, and fixes some indentation issues in the js.

### CODE REVIEW

**Comment by antgonza** [comment]
Depends on https://github.com/biocore/qiita/pull/1822, merge that one first

**Comment by antgonza** [comment]
Ready for review, thanks!

**Comment by ElDeveloper** [comment]
👍

File: qiita_db/util.py [hunk 1]
```diff
@@ -1391,11 +1391,12 @@ def generate_study_list(study_ids, build_samples):
 
             # shared with
             info['shared'] = []
-            for name, email in zip(info['shared_with_name'],
-                                   info['shared_with_email']):
-                if not name:
-                    name = email
-                info['shared'].append((email, name))
+            if info['shared_with_name'] and info['shared_with_email']:
```
Comment (josenavas) [inline qiita_db/util.py]: It looks like these lines of code where bugs introduced in the previous PR, which means that the tests do not capture this case. I recommend to add a specific test that will capture this case so it doesn't repeat.

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 2]
```diff
@@ -168,10 +174,16 @@ class SearchStudiesAJAX(BaseHandler):
     def get(self, ignore):
         user = self.get_argument('user')
         query = self.get_argument('query')
+        search_type = self.get_argument('search_type')
         echo = int(self.get_argument('sEcho'))
 
         if user != self.current_user.id:
             raise HTTPError(403, 'Unauthorized search!')
+        if search_type not in ['user', 'public']:
+            self.clear()
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: `raise HTTPError(400, 'Not a valid search type')`

File: qiita_pet/templates/list_studies.html [hunk 3]
```diff
@@ -21,7 +21,8 @@
 <script src="{% raw qiita_config.portal_dir %}/static/vendor/js/select2.min.js"></script>
 <script src="{% raw qiita_config.portal_dir %}/static/js/sharing.js"></script>
 <script type="text/javascript">
-var ajaxURL = "{% raw qiita_config.portal_dir %}/study/search/?&user={{current_user.id}}&sEcho=" + Math.floor(Math.random()*1001);
+var user_studies_ajaxURL = "{% raw qiita_config.portal_dir %}/study/search/?&user={{current_user.id}}&search_type=user&sEcho=" + Math.floor(Math.random()*1001);
```
Comment (josenavas) [inline qiita_pet/templates/list_studies.html]: Can we avoid using global variables? Specially this values are used only in a single place.

**Comment by josenavas** [comment]
Few comments. However, the sample selection doesn't work with this branch. Unsure where the bug is, but neither the `Add to analysis` nor the individual `Add` buttons for each biom table work.

File: qiita_db/util.py (hunk 1 above)
Comment (antgonza) [inline qiita_db/util.py]: k

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 2 above)
Comment (antgonza) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: The issue is that that error is logged in the DB and not shown in the GUI. Thus decided to go this route so it's shown.

File: qiita_pet/templates/list_studies.html (hunk 3 above)
Comment (antgonza) [inline qiita_pet/templates/list_studies.html]: yup, fixing.

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 2 above)
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: A user should never hit this error - this is more a safeguard for developers - or am I missing something?

Comment (antgonza) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: I was thinking in the restapi and that this should be the best method to access the list of studies.

Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: In that case you should be able to use the code above - is what I've been using in the qiita_db handlers. If the error message is not sent, then there is a problem in the function `write_error` on the base handler. I think the one in the qiita_db subdir is different and does it correctly.

**Pull request merged**